lib/services/nodes-api-service.js

```diff
@@ -509,6 +509,26 @@ function nodeApiServiceFactory(
     };
 
     /**
+     * Remove the tag from a list of nodes
+     * @param  {String}     tagName
+     * @return {Promise}
+     */
+    NodeApiService.prototype.masterDelTagById = function(tagName) {
```

Comment:
Thanks @uppalk1 , my understanding for tag is that it will be assigned a mongo unique identifier, but the tag name is not unique, because in  the database schema, there is not an unique flag for tag name like what we did for the injectableName of [taskDefinition] (https://github.com/RackHD/on-core/blob/master/lib%2Fmodels%2Ftask-definition.js#L24) , 
Do you mean the tag name should also be unique? If so we need an unique flag for tag schema.

Or the API is just to delete tag by identifier, but you named the argument with "tagName"? 

Anyway, my meaning is that keeping the API name and argument be consistent, both use `id` or both use `name`, then it will not confuse other developers.

Add @zyoung51